### DISCUSSION TITLE
Dynamic Routes Issue

### DISCUSSION THREAD

**Discussion by BedtimeAus** [vercel/next.js#24848] [Help]
Hi there

I have a catch all routes in my pages folder [...url].js

But the resolvedUrl in the page gives me the path and query parameters when I use Next Link

```
            <Link href={`${category.slug.current}`} as={`${category.slug.current}`} passHref >
              <a>
                {category.title}
              </Link>
            </Link>
```
ResolvedURL: "/kids-beds/single-beds?url=kids-beds&url=single-beds"

When I refresh the resolvedUrl looks fine:

"/kids-beds/single-beds"

What am I missing here?

TIA
Steve

**Comment by bobarros** [comment]
Are not missing this? You're using `${category.slug.current}` inside href and as.

So, it seems you have:

```jsx 
<Link href="/kids-beds/single-beds" as="/kids-beds/single-beds">
```

But you don't have a /kids-beds folder. 
So you need something like:

```jsx
<Link href="/kids-beds-single-beds" as="/kids-beds/single-beds">
```
